### PR TITLE
Split travel schema & document into own file

### PR DESCRIPTION
**Pull request by khmoran**
There are a few commits in this -- prob worth stepping through them to understand the large-ish file diff

Basically, I want to be able to have canonical full/minimal json examples for each schema so that other schemas that embed them can reuse them -- it'd be a pain to update a whole bunch of tests if one sub-document changes

### CODE REVIEW

**Comment by khmoran** [comment]
[Refactor dataserver model into classes, add tests](https://trello.com/c/9VunjXvq/133-refactor-dataserver-model-into-classes-add-tests)

File: data-serving/data-service/test/model/date-range.test.ts [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+    'DateRange',
+    dateRangeSchema,
+);
+
+describe('validate', () => {
+    it('an empty date range document is valid', async () => {
+        return new DateRange({}).validate();
+    });
+
+    it('a fully specified date range document is valid', async () => {
```
Comment (axmb) [inline data-serving/data-service/test/model/date-range.test.ts:16]: Minor thought: do we want a minimal (half-open) range?

File: data-serving/data-service/src/model/travel.ts [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+        location: locationSchema,
+        dateRange: dateRangeSchema,
+        purpose: {
+            type: String,
+            enum: Object.values(TravelPurpose),
+        },
+        additionalInformation: String,
+    },
+    {
+        strict: true,
```
Comment (axmb) [inline data-serving/data-service/src/model/travel.ts:23]: Do we have this specified `false` or `throw` at some top-level point? `true` is the default value

File: data-serving/data-service/test/model/travel.test.ts [hunk 3]
```diff
@@ -0,0 +1,31 @@
+import { TravelDocument, travelSchema } from '../../src/model/travel';
```
Comment (axmb) [inline data-serving/data-service/test/model/travel.test.ts:1]: Planning to json this, like the other ones?

File: data-serving/data-service/test/model/date-range.test.ts (hunk 1 above)
Comment (khmoran) [reply data-serving/data-service/test/model/date-range.test.ts:31]: done

File: data-serving/data-service/src/model/travel.ts (hunk 2 above)
Comment (khmoran) [reply data-serving/data-service/src/model/travel.ts]: nope -- removed (here & elsewhere)

File: data-serving/data-service/test/model/travel.test.ts (hunk 3 above)
Comment (khmoran) [reply data-serving/data-service/test/model/travel.test.ts:1]: sure -- I'll do it here

**Pull request merged**